assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * `core/modules` data store: module sharing settings
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { CORE_MODULES } from './constants';
+
+// Actions
+const SET_SHARING_MANAGEMENT = 'SET_SHARING_MANAGEMENT';
+const SET_SHARED_ROLES = 'SET_SHARED_ROLES';
+const RECEIVE_GET_SHARING_SETTINGS = 'RECEIVE_GET_SHARING_SETTINGS';
+
+const baseInitialState = {
+	sharingSettings: undefined,
+	savedSharingSettings: undefined,
+};
+
+const fetchSaveSharingSettingsStore = createFetchStore( {
+	baseName: 'saveSharingSettings',
+	controlCallback: ( { savedSharingSettings } ) => {
+		return API.set(
+			'core',
+			'modules',
+			'sharing-settings',
+			savedSharingSettings,
+			{
+				useCache: false,
+			}
+		);
+	},
+	reducerCallback: ( state, { settings } ) => {
+		return {
+			...state,
+			savedSharingSettings: settings,
+			sharingSettings: settings,
+		};
+	},
+	argsToParams: ( savedSharingSettings ) => ( { savedSharingSettings } ),
+	validateParams: ( { savedSharingSettings } = {} ) => {
+		invariant( savedSharingSettings, 'savedSharingSettings is required.' );
+	},
+} );
+
+const baseActions = {
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} moduleSlug Module slug.
+	 * @param {string} management New management role for module, one of: all_admins, owner.
+	 * @return {Object} Action for SET_SHARING_MANAGEMENT.
+	 */
+	setSharingManagement( moduleSlug, management ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		const managementRoles = [ 'all_admins', 'owner' ];
```

Comment:
Nit: this doesn't need to be defined inside the action as it never changes. Also, "roles" is a bit of a misnomer as these aren't really roles (as are used elsewhere) so that could be a bit confusing. How about a `validManagementValues` defined at the top?